libraries/AP_HAL_Linux/Util.h

```diff
@@ -113,6 +119,18 @@ class Util : public AP_HAL::Util {
     const char *custom_terrain_directory = nullptr;
     const char *custom_storage_directory = nullptr;
     static const char *_hw_names[UTIL_NUM_HARDWARES];
+
+#ifdef ENABLE_HEAP
+    struct heap_allocation_header {
+        size_t allocation_size; // size of allocated block, not including this header
+    };
+
+    struct heap {
+      size_t scripting_max_heap_size;
+      size_t current_heap_usage;
+    };
```

Comment:
this seems to be in the wrong abstraction layer...  If you need a "allocate_heap()" to be compatible with chibios/microcontrollers, then the HAL function should just allocate the memory in the right place and know nothing about scripting. Let the scripting part in AP_Script.